tests/unit/python/foglamp/services/common/test_services_common_utils.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import aiohttp
+import pytest
+from unittest import mock
+from unittest.mock import patch, MagicMock
+from foglamp.services.common import utils as utils
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("services", "common")
+class TestUtils:
+    async def test_ping_service_pass(self, loop, unused_port, mocker):
+        # GIVEN a service is running at a given URL
+        service = MagicMock()
+        service._protocol = 'http'
+        service._address = '0.0.0.0'
+        service._management_port = unused_port()
+
+        session = aiohttp.ClientSession(loop=loop)
+        mocker.patch.object(session, 'get', AsyncContextManagerMockForPing)
+        mocker.patch.object(utils, 'SESSION', session)
+
+        # WHEN the service is pinged with a valid URL
+        resp = await utils.ping_service(service)
+
+        # THEN ping response is received
+        assert resp is True
+
+    @pytest.mark.asyncio
+    async def test_ping_service_fail_bad_url(self, unused_port):
+        """This test will attempt 15 times before it returns False, hence it takes around 15-20 seconds to run."""
```

Comment:
its an integration test?

Reply:
Nope but the method does attempt 15 times and hence it takes time!

Reply:
we should ensure, correct endpoint is called, simulate the failure on first attempt and make sure next attempt is made. we should not let unit test wait for 15 seconds